misk/src/main/kotlin/misk/web/actions/ReadinessCheckAction.kt

```diff
@@ -40,9 +69,35 @@ class ReadinessCheckAction @Inject internal constructor(
       return Response("", statusCode = 503)
     }
 
-    val failedHealthChecks = healthChecks
-      .map { it.status() }
-      .filter { !it.isHealthy }
+    // Only null on first run
+    val (lastUpdate, statuses) = cachedStatus
+      // Block to get the initial status and enqueue background refresh
+      ?: refreshStatuses().also {
+        taskQueue.scheduleWithBackoff(Duration.ofMillis(config.readiness_refresh_interval_ms.toLong())) {
```

Comment:
Should we move this to separate `AbstractIdleService`? I worry that here we could schedule multiple job  (once every time this webaction is called.

Reply:
Agreed this should be moved into its own class

Reply:
This schedule action is only called when `cachedStatus == null`, which is only true on the first run... though I guess if two requests came in at the same time that could still happen.

Reply:
@tgregory-block Do you think an abstract service replaces the need for the `RepeatedTaskQueue`? I think it does, since it covers the run-as-single-thread benefit, but I'm curious what you think?

Reply:
Something in that class would still periodically need to run the refresh, so personally I'd stick the task queue in the abstract service. The abstract svc does get its own threads in `startup()` and `shutdown()` but they're not accessible after those methods return